src/SwedbankPay/Core/Library/OrderAction.php

```diff
@@ -322,20 +322,30 @@ public function refund($orderId, $amount = null, $vatAmount = 0)
 
         switch ($transaction['state']) {
             case 'Completed':
-                $this->updateOrderStatus(
-                    $orderId,
-                    OrderInterface::STATUS_REFUNDED,
-                    sprintf('Refunded: %s.', $amount),
-                    $transaction['number']
-                );
+                $info = $this->fetchPaymentInfo($paymentId);
+
+                if ((int) $info['payment']['remainingReversalAmount'] === 0) {
+                    $this->updateOrderStatus(
+                        $orderId,
+                        OrderInterface::STATUS_REFUNDED,
+                        sprintf('Refunded: %s. Transaction state: %s', $amount, $transaction['state']),
+                        $transaction['number']
+                    );
+                } else {
+                    $this->addOrderNote(
+                        $orderId,
+                        sprintf('Refunded: %s. Transaction state: %s', $amount, $transaction['state'])
+                    );
+                }
+
```

Comment:
These lines seem to repeat across the different payment types and actions, could it perhaps be put into a reusable function?

Reply:
@asbjornu It has repeated for invoice payments only. The checkout code has a different view (paymentOrder key instead of payment). I think it's non-important